test_util/launch_configs/README.md

```diff
@@ -0,0 +1,50 @@
+# DC/OS Launch Configuration YAML
+## Design Intention
+The intention of this configuration file is to provide an interface by which
+all deployments of DC/OS, regardless of provider have a similar format, thus
+complementing the goal of dcos-launch to provide a single tool for launching
+across a variety of provider APIs
+
+## Supported Deployments and Examples
+- [Simple AWS Cloudformation](aws-cf.yaml)
+- [Zen AWS Cloudformation](aws-zen-cf.yaml)
+- [Non-DC/OS AWS Linux Cluster](aws-linux-cluster.yaml)
+- [Onprem Install on AWS Linux Cluster](aws-onprem.yaml)
+- [Onprem Install on Previously Provisioned Linux Cluster](onprem-from-info.yaml)
+- [Azure Template Deployment](azure.yaml)
+
+## Keywords and Definitions
+
+### Required for Deployment
+* `deployment_type`: Can be one of `template`, `onprem`, or `cloud_cluster`.
+  * `template` refers to either an AWS cloudformation template or an Azure Resource Manager deployment template. Use of this options will require setting `deployment_name`, `template_url`, `provider`, and `provider_info`
+  * `cloud_cluster` will used hard-coded templates to deploy a cluster of machines in a shared network on the provider of choice (currently only supports AWS). `deployment_name` and `provider` must be set. See `cloud_cluster` below for options. Only `aws` provider is currently supported for this deploy option
+  * `onprem` refers to using the DC/OS installer (`dcos_generate_config.sh`) to deploy DC/OS onto a set of provided hosts. Those hosts can either be provisioned with keywords or by having previously created a `cloud_cluster` with dcos-launch. `installer_url` must be set as must `dcos_config` (see below)
+    * If `provider` is set to `cloud_cluster`, then the only other required field is `provider_info_path`, which points towards the cluster info JSON created by dcos-launch for which you want to install DC/OS
+    * If `provider` is not set to `cloud_cluster`, dcos-launch will use that provider to create the linux cluster and then immediately start deploying DC/OS to it (currently only supports `aws` provider). The linux cluster can be configured in the same manner as deploying a stand alone linux cluster (see `cloud_cluster` below), however the `instance_count` is omitted as it will be implied from `num_masters`, `num_public_agents`, and `num_private_agents`
```

Comment:
Can the two `provider` sections above be reworked or moved to underneath the provider section below? It was pretty confusing to me to show up here after the onprem provider details.